src/utils/async.ts

```diff
@@ -1,16 +1,39 @@
 /**
- * Maps over an AsyncIterator
+ * Maps results from an iterator and returns an iterator with new values
+ * @param iter the provided iterator
+ * @param f the functor
+ * @param mapReturn true if the return value must be mapped. Default: false
+ *
+ * @returns an aync iterator which values are mapped
  */
 export function mapAsyncIterator<T, V>(
   iter: AsyncIterator<T>,
-  f: (t: T) => V
+  f: (t: T) => V,
+  mapReturn?: false
```

Comment:
IMHO there's no need to map the return value. The map / filter semantic should follow the behavior of a `for of` loop (which stop reading before done is true)

Reply:
I agree, but it's actually legal to write this
```ts
function*(){
    yield 1;
    yield 2;
    return 3;
}
```

I'm just afraid we'll encounter cases in which we fail using map because of this

Reply:
that would be a bad usage of the iterator, I'd go for the simpler solution by now